### DISCUSSION TITLE
Save the session to a file

### DISCUSSION THREAD

**Discussion by ErikWil** [waf/CSharpRepl#315] [Ideas]
It would be great if there was a possibility to save a session to a .cxs file similar to the functionality found in the [IPython console] (https://gist.github.com/awesomebytes/4972b76459822845cc71)

**Comment by marckruzik** [comment]
I looked a bit into this subject. I'm writing notes for anyone interested.

The prompt history is saved in `"C:\Users\someone\AppData\Roaming\.csharprepl\prompt-history"`, with each prompt saved in Base64.
All sessions are there. To have only one session, a solution would be to record both the persistent history and a per-session history.
By the way, as the Prompt class managing the history is part of PrettyPrompt, so this is surely a feature that should be developed in the other repo.
https://github.com/waf/PrettyPrompt
